app/segments/(team)/chooseTeam.js

```diff
@@ -99,22 +104,56 @@ function ChooseTeam() {
           >
             <Button
               onPress={async () => {
-                //temporary, should be replaced with multiple team functionality
-                await setDoc(doc(db, "teams", "1", "users", currentUserId), {
-                  name: currentUserInfo["displayName"],
-                  // hardcoded pfp string for now, add pfp upload to profile settings in future PR
-                  pfp: "",
-                  // hardcoded "player" role for now, add role selection to profile settings in future PR
-                  role: "player",
-                  uid: currentUserId,
-                  assigned_data: [],
-                  uniqueDrills: [],
-                });
-                setCurrentUserId(currentUserId);
-                await invalidateMultipleKeys(queryClient, [
-                  ["userInfo", { userId: currentUserId }],
-                ]);
-                router.replace("/");
+                let completed = false;
+
+                const unregisterAuthObserver = onIdTokenChanged(
+                  auth,
+                  async (user) => {
+                    if (user && !completed) {
+                      await user.reload();
+                      if (user.emailVerified) {
+                        completed = true;
```

Comment:
IIRC the onAuthStateChanged in Auth.js only fires when you login or signup and not on token change. Do we need to set auth.currentUserVerified to true here? Maybe we'll need this variable in the code in the future?

Reply:
I didn't realize that context/Auth.js also has a an emailVerified property. Yeah, it should be better to combine them somehow (not sure).

Reply:
from my experience with displayName, signup.js sets the displayname on firebase but not locally, so I had to also set it locally. If you can use your onTokenChange observer to handle this it'll be great